exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -607,7 +607,14 @@ public long getRowCount() {
       fileSet = Sets.newHashSet(fileNames);
     }
 
-    if (fileNames.isEmpty()) {
+    List<String> finalFileNames;
+    if (fileSet != null) {
+      finalFileNames = Lists.newArrayList(fileSet);
```

Comment:
Actually, fileNames can contain duplicates whereas fileSet has a unique list.  fileSet was already present in the code earlier and used by removeUnneededRowGroups().  I am leveraging fileSet because when expanding directories it may be possible to end up with duplicate files which we don't want.   I could perhaps add a comment here.